src/tech/v3/dataset/column_index_structure.clj

```diff
@@ -0,0 +1,28 @@
+(ns tech.v3.dataset.column-index-structure
+  (:require [tech.v3.protocols.column :as col-proto]))
+
+(defn select-from-index
+  "Select a subset of the index as specified by the `mode` and
+  `selection-spec`. Each index-structure type supports different
+  modes. By default, tech.ml.dataset supports two index types:
+  `java.util.TreeMap` for non-categorical data, and `java.util.LinkedHashMap`
+  for categorical data.
+
+  Mode support for these types is as follows:
+
+  `java.util.TreeMap` - supports `::slice` and `::pick`.
+  `java.util.LinkedHashMap - support `::pick`
+
+  Usage by mode:
+
+  :slice
+  (select-from-index :slice {:from 2 :to 5})
+  (select-from-index :slice {:from 2 :from-inclusive? true :to 5 :to-inclusive? false})
+
+  :pick
+  (select-from-index :pick [:a :c :e])
```

Comment:
I considered setting a convention here of using fully qualified keywords, but after a [conversation with @joinr](https://clojurians.zulipchat.com/#narrow/stream/151168-clojure/topic/how.20to.20define.20namepace.20keywords.20in.20a.20ns/near/235923023) decided it may be better not to. The tech stack tends to use unqualified keywords and so breaking with that could be awkward. These keywords are also not packed into any data structure that is passed around to other systems so marking their provenance is possibly less important.

Reply:
Completely agreed.  I think if you are accepting datastructures that may existing in other systems then there is a strong argument for namespaced keywords.  For function arguments I personally do not like them.